sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ConfigurationUtils.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.ConfigurationPropertyBuilder;
+import com.azure.core.util.CoreUtils;
+
+import java.time.Duration;
+import java.util.function.Function;
+
+public class ConfigurationUtils {
+
+    private static final Function<String, Boolean> BOOLEAN_CONVERTER = (value) -> Boolean.parseBoolean(value);
```

Comment:
Why does the boolean converter return the primitive boolean while the integer converter returns the nullable Integer?

Reply:
Also, we should check `value` for null and return `null`, so, returning`Boolean` instead of `boolean` makes sense.